src/js/player.js

```diff
@@ -1601,6 +1601,88 @@ class Player extends Component {
     this.trigger('ratechange');
   }
 
+  /**
+   * Get or set vjs-waiting class on the player. This shows the loading spinner.
+   *
+   * @param  {boolean} [waiting]
+   *         - True if the player should be in the waiting state
+   *         - False otherwise[
+   *
+   * @param {boolean} [changeControls]
+   *        - Should the controls be enabled/disabled along with the waiting value?
+   *
+   * @return {boolean}
+   *         The current waiting state of the player
+   */
+  waiting(waiting, changeControls) {
+    if (typeof waiting === 'undefined') {
+      return this.hasClass('vjs-waiting');
+    }
+
+    // remove the current listner, since there will either be a new one
+    // or no need for one
+    if (this.waitingListener_) {
+      this.off('timeupdate', this.waitingListener_);
+      this.waitingListener_ = null;
+    }
+
+    // if waiting is false do the following and return
+    if (!waiting) {
+      this.removeClass('vjs-waiting');
+      if (changeControls) {
+        this.enableControls();
+      }
+      return;
+    }
+
+    // if waiting is true do the following
+
+    if (changeControls) {
+      this.disableControls();
+    }
+    this.addClass('vjs-waiting');
+
+    // Browsers may emit a timeupdate event after a waiting event. In order to prevent
+    // premature removal of the waiting class, wait for the time to change.
+    const timeWhenWaiting = this.currentTime();
+
+    this.waitingListener_ = () => {
+      if (timeWhenWaiting !== this.currentTime()) {
+        this.waiting(false, changeControls);
+      }
+    };
+
+    this.on('timeupdate', this.waitingListener_);
+  }
+
+  /**
+   * Disable playToggle and progress controls, effectively disabling playback
+   */
+  disableControls() {
+    if (!this.controlBar) {
+      return;
+    }
+    ['progressControl', 'playToggle'].forEach(function(k) {
+      if (this.controlBar[k]) {
+        this.controlBar[k].enable();
```

Comment:
Should this be `this.controlBar[k].disable();`?

Reply:
yeah